cellfinder/core/tools/IO.py

```diff
@@ -24,6 +24,40 @@ def get_tiff_meta(
 lazy_imread = delayed(imread)  # lazy reader
 
 
+def read_z_stack(path):
+    """
+    Reads z-stack, lazily, if possible.
+
+    If it's a text file or folder with 2D tiff files use dask to read lazily,
+    otherwise it's a single file tiff stack and is read into memory.
+
+    :param path: Filename of text file listing 2D tiffs, folder of 2D tiffs,
+        or single file tiff z-stack.
+    :return: The data as a dask/numpy array.
+    """
+    if path.endswith(".tiff") or path.endswith(".tif"):
+        tiff = TiffFile(path)
```

Comment:
Could you use ```with TiffFile(path) as tiff: ``` for this section? This will ensure the connection is closed correctly, as [they require in their docs](https://github.com/cgohlke/tifffile/blob/master/tifffile/tifffile.py#L4162)

Reply:
Added change!